### PR TITLE
Validate lexicographical ordering/uniqueness of message parents

### PR DESCRIPTION
**Pull request by Adam-Gleave**
# Description of change

Add a validation for lexicographical ordering/uniqueness of message parents.

Changes:
 - Implement check and relevant error for `Parents::unpack()` and `Parents::new()`.
 - Add tests for unsorted/not unique cases with both `unpack` and `new`.
 - Ensure random messages used for tests (in `bee-test/rand`) are sorted lexicographically by serialised form, and update tests to use this, so that they do not fail with the new error.

## Links to any relevant issues

Addresses issue iotaledger/bee#460.

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Run and passed all tests locally.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bee-message/src/parents.rs [hunk 1]
```diff
@@ -68,6 +72,6 @@ impl Packable for Parents {
             inner.push(MessageId::unpack(reader)?);
         }
 
-        Ok(Self(inner))
+        Ok(Self::new(inner)?)
```
Comment (thibault-martinez) [inline bee-message/src/parents.rs]:
```suggestion
        Self::new(inner)
```

File: bee-message/tests/parents.rs [hunk 2]
```diff
@@ -30,6 +31,22 @@ fn new_invalid_more_than_max() {
...
+    let mut inner = rand_message_ids(8);
+    inner.reverse();
+
+    assert!(matches!(Parents::new(inner.clone()), Err(Error::ParentsNotUniqueSorted)));
+}
+
+#[test]
+fn new_invalid_not_unique() {
+    let mut inner = rand_message_ids(7);
+    inner.push(inner[0]);
```
Comment (thibault-martinez) [inline bee-message/tests/parents.rs]: It's not straightforward that it tests unicity. Pushing the first element to the end also isn't sorted. Probably better to have two consecutive same elements ?

File: bee-message/src/parents.rs (hunk 1 above)
Comment (Adam-Gleave) [reply bee-message/src/parents.rs]: Good catch – thanks, should be addressed now.

File: bee-message/tests/parents.rs (hunk 2 above)
Comment (Adam-Gleave) [reply bee-message/tests/parents.rs]: I've updated to repeat the last element instead, so all IDs will be sorted, just with one repeated element.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**